src/frontend/src/expr/table_function.rs

```diff
@@ -226,6 +229,132 @@ impl TableFunction {
+                    }
+                    Some(Err(err)) => {
+                        return Err(err);
+                    }
+                    None => {
+                        return Err(BindError(
+                            "table_scan function only accepts constant arguments".to_string(),
+                        )
+                        .into());
+                    }
+                }
+            }
+            evaled_args
+        };
+
+        #[cfg(madsim)]
+        {
+            return Err(crate::error::ErrorCode::BindError(
+                "table_scan can't be used in the madsim mode".to_string(),
+            )
+            .into());
+        }
+
+        #[cfg(not(madsim))]
+        {
+            let schema = tokio::task::block_in_place(|| {
+                RUNTIME.block_on(async {
+                    let (client, connection) = tokio_postgres::connect(
+                        format!(
+                            "host={} port={} user={} password={} dbname={}",
+                            evaled_args[0],
+                            evaled_args[1],
+                            evaled_args[2],
+                            evaled_args[3],
+                            evaled_args[4]
+                        )
+                        .as_str(),
+                        tokio_postgres::NoTls,
+                    )
+                    .await?;
+
+                    tokio::spawn(async move {
+                        if let Err(e) = connection.await {
+                            tracing::error!(
+                                "postgres_query_executor: connection error: {:?}",
+                                e.as_report()
+                            );
+                        }
+                    });
+
+                    let statement = client.prepare(evaled_args[5].as_str()).await?;
+
+                    let mut rw_types = vec![];
+                    for column in statement.columns() {
+                        let name = column.name().to_string();
+                        let data_type = match *column.type_() {
+                            TokioPgType::BOOL => DataType::Boolean,
+                            TokioPgType::INT2 => DataType::Int16,
+                            TokioPgType::INT4 => DataType::Int32,
+                            TokioPgType::INT8 => DataType::Int64,
+                            TokioPgType::FLOAT4 => DataType::Float32,
+                            TokioPgType::FLOAT8 => DataType::Float64,
+                            TokioPgType::NUMERIC => DataType::Decimal,
+                            TokioPgType::DATE => DataType::Date,
+                            TokioPgType::TIME => DataType::Time,
+                            TokioPgType::TIMESTAMP => DataType::Timestamp,
+                            TokioPgType::TIMESTAMPTZ => DataType::Timestamptz,
+                            TokioPgType::TEXT => DataType::Varchar,
+                            TokioPgType::BYTEA => DataType::Bytea,
```

Comment:
This list is not in sync with the one handled in executor:
* Missing interval and jsonb
* Both `text` and `varchar` can be mapped to `varchar`